drivers/apcgalaxy5500-mib.c

```diff
@@ -0,0 +1,299 @@
+ *   '.1.3.6.1.4.1.318.1.3.4.4': ApcMSP
+ *  - from Bill Seligman
+ *   .1.3.6.1.4.1.318.1.3.2.11
+ *   .1.3.6.1.4.1.318.1.3.2.12
+ *   .1.3.6.1.4.1.318.1.3.27
+ *   .1.3.6.1.4.1.318.1.3.2.7
+ *   .1.3.6.1.4.1.318.1.3.2.8
+ */
+
+/* TODO: find the right sysOID for this MIB
+ * Ie ".1.3.6.1.4.1.318.1.1.1" or ".1.3.6.1.4.1.318" or? */
+
+/*     .1.3.6.1.4.1.318.1.1.1
+ *       enterprise^
+ *       apc ---------^
+ *       products ------^
+ *       hardware --------^
+ *       ups ---------------^
+ *  ref: ftp://ftp.apc.com/apc/public/software/pnetmib/mib/404/powernet404.mib
+ */
+
+/* info elements */
+
+
+
+#define APCC_G5500_OID_BATT_STATUS	".1.3.6.1.4.1.318.1.1.1.2.1.1.0"
+/* Defines for APCC_G5500_OID_BATT_STATUS */
+static info_lkp_t APCC_G5500_batt_info[] = {
+	{ 1, "", NULL, NULL },	/* unknown */
+	{ 2, "", NULL, NULL },	/* batteryNormal */
+	{ 3, "LB", NULL, NULL },	/* batteryLow */
+	{ 0, NULL, NULL, NULL }
+} ;
+
+#define APCC_G5500_OID_POWER_STATUS	".1.3.6.1.4.1.318.1.1.1.11.1.1.0"
+/* Defines for APCC_G5500_OID_POWER_STATUS */
+static info_lkp_t APCC_G5500_pwr_info[] = {
+   { 0, "OL", NULL, NULL},
+   { 1, "OB", NULL, NULL}
+} ;
+
+#define APCC_G5500_OID_CAL_RESULTS	".1.3.6.1.4.1.318.1.1.1.7.2.6.0"
+static info_lkp_t APCC_G5500_cal_info[] = {
+    { 1, "", NULL, NULL },          /* Calibration Successful */
+    { 2, "", NULL, NULL },          /* Calibration not done, battery capacity below 100% */
+    { 3, "CAL", NULL, NULL },       /* Calibration in progress */
+    { 0, NULL, NULL, NULL }
+};
+
+#define APCC_G5500_OID_NEEDREPLBATT	".1.3.6.1.4.1.318.1.1.1.2.2.4.0"
+static info_lkp_t APCC_G5500_battrepl_info[] = {
+    { 1, "", NULL, NULL },          /* No battery needs replacing */
+    { 2, "RB", NULL, NULL },        /* Batteries need to be replaced */
+    { 0, NULL, NULL, NULL }
+};
+
+#define APCC_G5500_OID_TESTDIAGRESULTS ".1.3.6.1.4.1.318.1.1.1.7.2.3.0"
+static info_lkp_t APCC_G5500_testdiag_results[] = {
+    { 1, "Ok", NULL, NULL },
+    { 2, "Failed", NULL, NULL },
+    { 3, "InvalidTest", NULL, NULL },
+    { 4, "TestInProgress", NULL, NULL },
+    { 0, NULL, NULL, NULL }
+};
+
+#define APCC_G5500_OID_SENSITIVITY ".1.3.6.1.4.1.318.1.1.1.5.2.7.0"
+static info_lkp_t APCC_G5500_sensitivity_modes[] = {
+    { 1, "auto", NULL, NULL },
+    { 2, "low", NULL, NULL },
+    { 3, "medium", NULL, NULL },
+    { 4, "high", NULL, NULL },
+    { 0, NULL, NULL, NULL }
+};
+
+#define APCC_G5500_OID_TRANSFERREASON "1.3.6.1.4.1.318.1.1.1.3.2.5.0"
+static info_lkp_t APCC_G5500_transfer_reasons[] = {
+    { 1, "noTransfer", NULL, NULL },
+    { 2, "highLineVoltage", NULL, NULL },
+    { 3, "brownout", NULL, NULL },
+    { 4, "blackout", NULL, NULL },
+    { 5, "smallMomentarySag", NULL, NULL },
+    { 6, "deepMomentarySag", NULL, NULL },
+    { 7, "smallMomentarySpike", NULL, NULL },
+    { 8, "largeMomentarySpike", NULL, NULL },
+    { 9, "selfTest", NULL, NULL },
+    { 10, "rateOfVoltageChange", NULL, NULL }
```

Comment:
Here and above, lookup tables must end with a `{ 0, NULL, NULL, NULL }` entry to stop iterations.

Reply:
Didn't think of that, this code snippet is copied from `drivers/apc-mib.c`